src/main/java/com/netflix/imflibrary/RESTfulInterfaces/IMPValidator.java

```diff
@@ -593,7 +594,10 @@ public static Long getRandomIndexPackSize(PayloadRecord essenceFooter4Bytes){
                     imfErrorLogger);
 
                 MXFOperationalPattern1A.HeaderPartitionOP1A headerPartitionOP1A = MXFOperationalPattern1A.checkOperationalPattern1ACompliance(headerPartition, imfErrorLogger);
-                IMFConstraints.checkIMFCompliance(headerPartitionOP1A, imfErrorLogger);
+                IMFConstraints.HeaderPartitionIMF headerPartitionIMF = IMFConstraints.checkIMFCompliance(headerPartitionOP1A, imfErrorLogger);
+                if (headerPartitionIMF.hasMatchingEssence(HeaderPartition.EssenceTypeEnum.IABEssence)) {
```

Comment:
I don't think this `hasMatchingEssence` check is working the way you intended.

Internally, it just maps the `EssenceTypeEnum` to a `MXFDataDefinition`, and then compares those values. But that mapping is not one-to-one. For IAB `EssenceTypeEnum.IABEssence` -> `MXFDataDefinition.SOUND` But also, for other audio essence, `EssenceTypeEnum.MainAudioEssence` -> `MXFDataDefinition.SOUND`.

As a result, providing a _non_-IAB audio essence still passes the `hasMatchingEssence` check against IABEssence. This isn't a big problem, because `IABTrackFileConstraints.checkCompliance` correctly recognizes the essence as not IAB, and exits gracefully. But I don't think that was how you wanted this check to work.

Reply:
I changed the test to`headerPartitionIMF.getEssenceType() == HeaderPartition.EssenceTypeEnum.IABEssence`.